### PR TITLE
feat(AvatarGroup): adds AvatarGroup component

### PR DESCRIPTION
**Pull request by riyalohia**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/avatar/Avatar.tsx [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+
+export interface CompProps extends BaseProps {
   /**
    * Color of the `Avatar`
-   * @default "primary"
    */
   appearance?: Appearance;
   /**
-   * **Only first 2 characters are rendered**
+   * **Only first 2 characters are rendered** **(SOON TO BE DEPRECATED)**
```
Comment (aditya-kumawat) [inline core/components/atoms/avatar/Avatar.tsx]: **Only first 2 characters are rendered (SOON TO BE DEPRECATED)**

File: core/components/atoms/avatarGroup/AvatarGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+interface PopperProps {
+  popperRenderer?: (names: AvatarData[]) => JSX.Element;
+  appendToBody?: boolean;
+  dark?: boolean;
+  position?: Position;
+  on?: ActionType;
+  maxHeight?: number;
+  popperClassName?: string;
```
Comment (aditya-kumawat) [inline core/components/atoms/avatarGroup/AvatarGroup.tsx:23]: Can we rename it from `popperClassName` to `className`?

File: core/components/atoms/avatarGroup/AvatarGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+interface CompProps extends BaseProps {
+  /**
+   * List of `Avatars`
+   *
+   * <pre className="DocPage-codeBlock">
+   *
```
Comment (aditya-kumawat) [inline core/components/atoms/avatarGroup/AvatarGroup.tsx]: Extra lines not required in `<pre>` block.

**Comment by aditya-kumawat** [comment]
@riyalohia Please add `position` prop to configure position of Tooltip for Avatars.

File: core/components/atoms/avatarGroup/AvatarGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+   * }
+   *
+   * | Name | Description | Default |
+   * | --- | --- | --- |
+   * | popperRenderer | Callback function to create custom popover content  | |
+   * | appendToBody | Appends `Popover` wrapper inside body | true |
+   * | dark | Changes background of `Popover` | true |
+   * | position | Position to place `Popover` | bottom |
+   * | maxHeight | Max height of `Popover Text Wrapper` (does not work in case of custom popperRenderer) | 150 |
+   * | popperClassName | Custom classname added to `Popover` | |
```
Comment (aditya-kumawat) [inline core/components/atoms/avatarGroup/AvatarGroup.tsx]: Description of `on` missing

File: core/components/atoms/avatarGroup/AvatarGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  } = popoverOptions;
+
+  const baseProps = extractBaseProps(props);
+
+  const extraAvatars = list.length > max ? list.length - max : 0;
+
+  const style = {
+    borderRadius: '50%',
+    backgroundColor: `${borderColor}`,
+    border: `2px solid ${borderColor}`,
```
Comment (aditya-kumawat) [inline core/components/atoms/avatarGroup/AvatarGroup.tsx]: use variable for 2px

File: core/components/atoms/avatar/Avatar.tsx [hunk 6]
```diff
@@ -37,6 +38,10 @@ export interface AvatarProps extends BaseProps {
    * Determines if tooltip is visible
    */
   withTooltip: boolean;
+  /**
+   * Position to place the tooltip
+   */
+  tooltipPosition: Position;
```
Comment (aditya-kumawat) [inline core/components/atoms/avatar/Avatar.tsx]: replace with TooltipProps['position']

**Review by aditya-kumawat — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**